numba/targets/arrayobj.py

```diff
@@ -905,6 +905,58 @@ def array_argmax_impl(arry):
     return context.compile_internal(builder, array_argmax_impl, sig, args)
 
 
+@builtin
+@implement(numpy.median, types.Kind(types.Array))
+@implement("array.median", types.Kind(types.Array))
+def array_median(context, builder, sig, args):
+    def partition(A, low, high):
+        # median of three {low, middle, high}
+        middle = (low+high) // 2
+        if A[middle] > A[high]:
+            A[middle], A[high] = A[high], A[middle]
```

Comment:
There's no need to mutate the array for the median-of-three computation, you could simply use some local variables.